jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/CachingHttpContentFactory.java

```diff
@@ -0,0 +1,571 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.time.Instant;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.eclipse.jetty.io.ByteBufferPool;
+import org.eclipse.jetty.io.NoopByteBufferPool;
+import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.NanoTime;
+import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.resource.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>
+ * {@link HttpContent.Factory} implementation that wraps any other {@link HttpContent.Factory} instance
+ * using it as a caching authority. Only HttpContent instances whose path is not a directory are cached.
+ * </p>
+ * <p>
+ * No eviction is done by this {@link HttpContent.Factory}, once an entry is in the cache it is always
+ * assumed to be valid. This class can be extended to implement the validation behaviours on
+ * {@link CachingHttpContent} which allow entries to be evicted once they become invalid.
+ * </p>
+ * @see ValidatingCachingContentFactory
+ */
+public class CachingHttpContentFactory implements HttpContent.Factory
+{
+    private static final Logger LOG = LoggerFactory.getLogger(CachingHttpContentFactory.class);
+
+    private final HttpContent.Factory _authority;
+    private final ConcurrentMap<String, CachingHttpContent> _cache = new ConcurrentHashMap<>();
+    private final AtomicLong _cachedSize = new AtomicLong();
+    private final ByteBufferPool _byteBufferPool;
+    private int _maxCachedFileSize = 128 * 1024 * 1024;
```

Comment:
Those default values should be javadoc'ed.

Reply:
done